cdap-app-templates/cdap-program-report/src/main/java/co/cask/cdap/report/ReportGenerationSpark.java

```diff
@@ -179,11 +197,15 @@ public void getReportStatus(HttpServiceRequest request, HttpServiceResponder res
       try {
         reportGenerationInfo = getReportGenerationInfo(reportId, reportIdDir);
       } catch (Exception e) {
-        LOG.error("Failed to get report generation info for report with id {}.", reportId, e);
-        responder.sendError(500, String.format("Failed to get report generation info for report with id %s" +
+        LOG.error("Failed to get the status for report with id {}.", reportId, e);
+        responder.sendError(500, String.format("Failed to get the status for report with id %s" +
                                                  " because of error: %s", reportId, e.getMessage()));
         return;
       }
+      if (ReportStatus.DELETED.equals(reportGenerationInfo.getStatus())) {
+        responder.sendError(404, String.format("Report with id %s does not exist.", reportId));
```

Comment:
rather say, it has expired.

Reply:
I think in a user's perspective, an expired report and a deleted report are essentially the same, since both cannot be listed, read, or checked for status, and both cannot be be brought back. Also the a report with status can also mean that the report is deleted by the user, so it will be confusing to say that this report has expired